CMakeLists.txt

```diff
@@ -81,9 +84,23 @@ add_definitions (-DBOOST_ASIO_DISABLE_BOOST_DATE_TIME=1)
 add_definitions (-DBOOST_ASIO_DISABLE_BOOST_REGEX=1)
 add_definitions (-DBOOST_COROUTINES_NO_DEPRECATION_WARNING=1)
 
-include_directories (${BOOST_ROOT})
+OPTION(FIND_BOOST "Build beast against an existing Boost install, rather than in-tree")
+if (FIND_BOOST)
+  set(Boost_USE_STATIC_LIBS ON)
+  find_package(Boost 1.66.0 REQUIRED COMPONENTS system filesystem coroutine)
+  include_directories(${Boost_INCLUDE_DIRS})
```

Comment:
Usage requirements should be applied to a target.

Reply:
I feel a lot of your comments here are directed at the code as it already stood, rather than this pull request, and probably should be handled in a separate (possibly preceding) change.

For this series, I didn't want to rewrite the CMake build system, simply extend what was already there to accommodate a different workflow. @vinniefalco has expressed reluctance to take on the workload of a full CMake-based build system, given the minimal benefits it provides to him as a maintainer, and almost-zero benefits it provides to users of this library (who would get it via Boost and hence never interact with this build-system at all).

Similarly, I wasn't particularly interested in trying to full CMake-ify this code, for the same reasons. If and when BCM or similar lands in Boost, then it's worth revisiting this, as the potential maintainer load would be much lower, and the user-visible benefits much higher.